docs/dev-environment.md

```diff
@@ -0,0 +1,46 @@
+## Development Environment
```

Comment:
I think this file looks fine as it is; probably doesn't really need a big change (or any)